tests/unit-tests/UnitTests/Services/BookingServiceTests.cs

```diff
@@ -78,28 +81,124 @@ public BookingServiceTests()
               _mockHttpContextAccessor.Object);
         }
 
+        [Fact]
+        public async Task Get_Should_CallMapAppointmentId_WhenAppointmentType_HasAppointmentIdKey()
+        {
+            // Arrange
+            var questionId = "bookingQuestion";
+            var appointmentId = new Guid("022ebc92-1c51-4a68-a079-f6edefc63a07");
+            string sessionGuid = "session";
+
+            BookingInformation cachedBookingInfo = new() { AppointmentTypeId = appointmentId };
+            _mockDistributedCache.Setup(_ => _.GetString(sessionGuid))
+                .Returns(JsonConvert.SerializeObject(new FormAnswers
+                {
+                    FormData = new Dictionary<string, object> { { $"{BookingConstants.APPOINTMENT_TYPE_SEARCH_RESULTS}", cachedBookingInfo } }
+                }));
+
+            var element = new ElementBuilder()
+                .WithType(EElementType.Booking)
+                .WithQuestionId(questionId)
+                .WithAppointmentType(new AppointmentType
+                {
+                    Environment = environmentName,
+                    AppointmentIdKey = questionId
+                })
+                .Build();
+
+            var page = new PageBuilder().WithElement(element).Build();
+
+            // Act
+            try
+            {
+                _ = await _service.Get("form", page, sessionGuid);
+            }
+            catch
+            {
+                // Assert
+                _mockMappingService.Verify(mappingService =>
+                    mappingService.MapAppointmentId(It.IsAny<AppointmentType>(), It.IsAny<FormAnswers>()), Times.Once);
+            }
```

Comment:
Should these be throwing an exception? if so u can use Assert.ThrowsAsync<Exception>(), To verify it is throwing an exception.

Reply:
Not checking exception. Moved Verify to Finally block.

Reply:
Added code to make the tests look nicer